code/modules/roguetown/roguemachine/hoardmaster.dm

```diff
@@ -119,12 +128,126 @@
+			var/unlock_time = SSticker.round_start_time + PA.time_lock
+			if(world.time < unlock_time) // Not enough time has passed
+				contents += "[PA.name] (Locked - Available in [time2text(unlock_time - world.time, "hh:mm")])<BR>"
+			else // Item is available for purchase
+				contents += "[PA.name] [PA.contains.len > 1 ? "x[PA.contains.len]" : ""] - ([PA.cost])<a href='?src=[REF(src)];buy=[PA.type]'>BUY</a><BR>"
 
 	var/datum/browser/popup = new(user, "HOARDMASTER", "", 370, 600)
 	popup.set_content(contents)
 	popup.open()
 
+/obj/structure/roguemachine/Hoardmaster/buckle_mob(mob/living/carbon/human/H, force = FALSE, check_loc = TRUE)
+	if (!istype(H, /mob/living/carbon/human))
+		to_chat(usr, span_warning("The Hoardmaster is not interested in this."))
+		return FALSE
+
+	if(HAS_TRAIT(H, TRAIT_COMMIE))
+		to_chat(usr, span_warning("The Hoardmaster does not hunger enough to devour one of its own."))
+		return FALSE
+
+	if(HAS_TRAIT(H, TRAIT_MATTHIOS_BRAND) || HAS_TRAIT(H, TRAIT_MATTHIOS_BRAND_OLD))
+		to_chat(usr, span_warning("The Hoardmaster has already humiliated this one."))
+		return FALSE
+
+	if(!H.mind)
+		to_chat(usr, span_warning("The Hoardmaster seeks to subjugate ones that are aware."))
+		return FALSE
+
+	return ..(H, force, FALSE)
+
+/obj/structure/roguemachine/Hoardmaster/unbuckle_mob(mob/living/user)
+	if(subjugating)
+		to_chat(usr, span_warning("The Hoardmaster will not be interrupted."))
+		return FALSE
+	else
+		..()
+
+/obj/structure/roguemachine/Hoardmaster/attack_right(mob/living/carbon/human/user)
+	set waitfor = FALSE
+
+	if(!user || !ishuman(user)) 
+		return
+
+	if(!HAS_TRAIT(user, TRAIT_COMMIE))
+		to_chat(user, span_notice("The Hoardmaster does not acknowledge you."))
+		return
+
+	var/mob/living/carbon/human/H = null
+	for(var/l in buckled_mobs)
+		H = l
+	if(!H)
+		to_chat(user, span_notice("The Hoardmaster's maw is empty."))
+		return
+	if(!H.buckled)
+		to_chat(user, span_notice("The Deposed is not secure."))
+		return
+	if(H.stat == DEAD)
+		to_chat(user, span_notice("The Deposed is defunct."))
+		return
+
+	if(!do_after(user, 3 SECONDS, TRUE, H))
+		return
+
+	playsound(src.loc, 'sound/items/pickgood1.ogg', 100, TRUE, -1)
+	H.Paralyze(20 SECONDS)
+
+	subjugating = TRUE
+	say("AN ACCEPTABLE OFFERING.")
+
+	var/payout = 400
+
+	sleep(2 SECONDS)
+
+	playsound(src.loc, 'sound/items/beartrap.ogg', 100, TRUE, -1)
+	visible_message(span_warning("The Hoardmaster's maw closes around [H]'s head!"))
+
+	sleep(2 SECONDS)
+
+	visible_message(span_warning("Jagged teeth find purchase on [H]'s temple!"))
+	H.emote("scream")
+	if(HAS_TRAIT(H, TRAIT_NOBLE))
+		say("NOW YOU WILL KNOW TRUE SUBJUGATION.")
+		payout = 800
+	else
+		say("A MARK OF SHAME.")
+	ADD_TRAIT(H, TRAIT_MATTHIOS_BRAND, TRAIT_GENERIC)
+	H.apply_status_effect(/datum/status_effect/matthiosbrand)
+
+	sleep(3 SECONDS)
+
+	visible_message(span_warning("The Dragon begins to swallow [H] whole!"))
```

Comment:
I really, Really wish we had a better method then 'eating' the person who's being ransomed. Especially with our demographics is not a good look >_>;;